src/api/spec/models/comment_spec.rb

```diff
@@ -0,0 +1,86 @@
+require "rails_helper"
+
+RSpec.shared_examples "a comment" do
+  let(:comment_type) { described_class.name.underscore }
+  let(:comment) { create(comment_type) }
+
+  describe "A comment" do
+    it { expect(build(comment_type)).to be_valid }
+  end
+
+  describe "Comment associations" do
+    it { is_expected.to belong_to(:bs_request).inverse_of(:comments) }
+    it { is_expected.to belong_to(:project).inverse_of(:comments) }
+    it { is_expected.to belong_to(:package).inverse_of(:comments) }
+    it { is_expected.to belong_to(:user).inverse_of(:comments) }
+
+    it { is_expected.to have_many(:children) }
```

Comment:
why not 

`it { is_expected.to have_many(:children).dependent(:destroy).class_name('Comment').with_foreign_key('parent_id') }`

http://www.rubydoc.info/github/thoughtbot/shoulda-matchers/Shoulda%2FMatchers%2FActiveRecord%3Ahave_many